### ISSUE TITLE
git should throw FileNotFound or similar FS-error when failing to read a file

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#155933]
There is [this](https://github.com/microsoft/vscode/blob/adee8fa29ad3635418ff995b235c4b08a40c8434/extensions/git/src/fileSystemProvider.ts#L190) which makes non-existent files appear as the empty string. This causes headaches for the 3wm-editor because we cannot differentiate between the case of empty and missing/deleted/merged files

This can be reproduce by typing `git merge --abort` while the merge editor is open

**Comment by joaomoreno** [comment]
It's actually going to break other things in the diff editor and quick diff feature. Needs real testing to investigating if there are other ways to accomplish what the empty string is currently accomplishing across the workbench features.

**Comment by lszomoru** [comment]
There have been some issues related to quick diff that I addressed by pushing a fix for this issue -  https://github.com/microsoft/vscode/pull/236676. I have validated that the quick diffs are working as expected following the change. I will do more testing around the diff editor for the working tree and the index focusing on new/deleted/ignored files as these were the files that used to be returned as an empty file. I would like to ask you to keep an eye out as well as let me know if you see any issues. Thanks!
